clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -5061,14 +5061,23 @@ bool Util::isSyclHalfType(QualType Ty) {
 }
 
 bool Util::isSyclSpecConstantType(QualType Ty) {
-  std::array<DeclContextDesc, 5> Scopes = {
+  std::array<DeclContextDesc, 6> Scopes = {
+      Util::MakeDeclContextDesc(Decl::Kind::Namespace, "cl"),
+      Util::MakeDeclContextDesc(Decl::Kind::Namespace, "sycl"),
+      Util::MakeDeclContextDesc(Decl::Kind::Namespace, "ext"),
+      Util::MakeDeclContextDesc(Decl::Kind::Namespace, "oneapi"),
+      Util::MakeDeclContextDesc(Decl::Kind::Namespace, "experimental"),
+      Util::MakeDeclContextDesc(Decl::Kind::ClassTemplateSpecialization,
+                                "spec_constant")};
+  std::array<DeclContextDesc, 5> ScopesDeprecated = {
```

Comment:
Should we add a warning for using deprecated  namespace @AaronBallman @premanandrao ?

Reply:
Yes; as a rule of thumb, when we deprecate something we should alert users when they're using it. Otherwise the deprecation doesn't help us to remove anything (we'll just silently break their code, which we could do without bothering to have a deprecation period at all).

Reply:
We have deprecation warnings. See e.g., https://github.com/intel/llvm/pull/4014/files#diff-75fc8ffcd0922ca67040695dbb6811aacc406812b55436b29f171cbac2e44c92R283

Reply:
Ok. Thank you for confirmation